src/DotNet.Testcontainers/Builders/AbstractBuilder.cs

```diff
@@ -1,7 +1,8 @@
-﻿namespace DotNet.Testcontainers.Builders
+namespace DotNet.Testcontainers.Builders
 {
   using System;
   using System.Collections.Generic;
+  using Docker.DotNet;
```

Comment:
The library (builder) does not expose models from this namespace. Please see the discussions in #370 too. I think it makes sense to combine the endpoint and credential in a single interface and offer different implementations specific for the authentication.